attrs.go

```diff
@@ -172,75 +171,3 @@ func marshalFileInfo(b []byte, fi os.FileInfo) []byte {
 
 	return b
 }
-
-// toFileMode converts sftp filemode bits to the os.FileMode specification
-func toFileMode(mode uint32) os.FileMode {
-	var fm = os.FileMode(mode & 0777)
-	switch mode & S_IFMT {
-	case syscall.S_IFBLK:
-		fm |= os.ModeDevice
-	case syscall.S_IFCHR:
-		fm |= os.ModeDevice | os.ModeCharDevice
-	case syscall.S_IFDIR:
-		fm |= os.ModeDir
-	case syscall.S_IFIFO:
-		fm |= os.ModeNamedPipe
-	case syscall.S_IFLNK:
-		fm |= os.ModeSymlink
-	case syscall.S_IFREG:
-		// nothing to do
-	case syscall.S_IFSOCK:
-		fm |= os.ModeSocket
-	}
-	if mode&syscall.S_ISGID != 0 {
```

Comment:
Huh… one would think that these could be set as (for example) `syscall.S_ISGID = 0` and then this would be recognized as always false, and thus optimized out for us…

I‘m sure there are subtle reasons why they chose to do so.

Reply:
The `syscall` package is not meant to be portable. It's actually currently frozen, so I think don't we can add `S_ISGID` for plan9 even if we wanted to. In fact, the `syscall` package might be [removed](https://github.com/golang/go/issues/5616#issuecomment-349783401) in the furture in favor of `golang.org/x/sys`.

Reply:
Indeed. And that‘s probably a good thing. But it‘s weird that they stopped at extended compatibility for windows and JS, but not for plan9.

Maybe it lost priority, and didn‘t keep up? Not sure. Just seems weird in my head to make all the fictional values for windows and JS, but not for plan9. :laughing: